src/apis/resource/resource_api.py

```diff
@@ -85,114 +81,42 @@ def get(self, identifier, cat_type="program"):
-
-    def _get_lod_resource(
-        self, level, identifier, mime_type, accept_profile, app_config
-    ):
-        """ Generates the expected data based on the mime_type.
-            :param level: meaning the catalogue type: 'program' (default), 'series', 'season', 'scene'.
-            :param identifier: the DAAN id.
-            :param mime_type: the mime_type, or serialization the resource is requested in.
-            :param accept_profile: the profile (or model/schema/ontology) the data is requested in. \
-                See: https://www.w3.org/TR/dx-prof-conneg/#related-http
-                Example: Accept: application/ld+json; profile="http://schema.org"
-            :param app_config: the application configuration
-            :return: RDF data in a response object
-        """
-        mt = None
-        try:
-            mt = MimeType(mime_type)
-        except ValueError:
-            logger.info(
-                f"Given mime type cannot be used. Fall back to default mime type {MimeType.JSON_LD.to_ld_format()}."
-            )
-            mt = MimeType.JSON_LD
-
-        profile = get_profile_by_uri(accept_profile, app_config)
-        profile_prefix = profile["prefix"]
         logger.info(
-            f"Getting requested resource with level: '{level}' and '{identifier}' from the flex store using profile '{profile_prefix}'."
+            f"Getting the RDF in the proper serialization format for {lod_url}."
         )
-
-        storage_handler_class = import_class(profile["storage_handler"])
-        resp, status_code, headers = storage_handler_class(
-            app_config, profile
-        ).get_storage_record(level, identifier, mt.to_ld_format())
-
-        # make sure to apply the correct mimetype for valid responses
-        if status_code == 200:
-            logger.info("Valid response from the flex data store.")
-            headers = {"Content-Type": mt.value}
-            if profile.get("uri") is not None:
-                content_profile = profile.get("uri")
-                headers["Content-Profile"] = content_profile
-            logger.info(
-                "Return requested data in the required serialization format and profile."
+        rdf_graph = util.ld_util.get_lod_resource_from_rdf_store(
+            lod_url,
+            current_app.config.get("SPARQL_ENDPOINT"),
+            current_app.config.get("URI_NISV_ORGANISATION"),
+        )
+        if rdf_graph is not None:
+            serialised_graph = rdf_graph.serialize(
+                format=mime_type.to_ld_format(), auto_compact=True
             )
-            return Response(resp, mimetype=mt.value, headers=headers)
-        elif status_code == 403:
-            logger.error(f"{status_code} {resp}")
-            return APIUtil.toErrorResponse("access_denied")
-        elif status_code == 404:
-            logger.error(f"{status_code} {resp}")
-            return APIUtil.toErrorResponse("not_found")
```

Comment:
This is because the resource_api itself doesn't do the retrieval any more. That goes via the sparql_construct_query() in the existing ld_util.py code, which is not set up to look at status. Do we want to refactor that function in this issue, or open a new issue for it?

Reply:
rather open a new issue for it. 
In that issue we might then address as well that we actually should serve some page with "this resource used to exist but has passed away" tombstone message for deleted resources.

Reply:
good plan. I'll open an issue and mention the tombstone as well

Reply:
https://github.com/beeldengeluid/beng-lod-server/issues/488

Reply:
>good plan. I'll open an issue and mention the tombstone as well
Sorry, I just did, but please edit if you feel like it.

Reply:
https://github.com/beeldengeluid/beng-lod-server/issues/489